### PR TITLE
Add the ability to fail builds in case if code coverage was decreased in a change request build

### PR DESCRIPTION
**Pull request by tempora-mutantur**
Hi

This is how this looks:

image

image

Thank you.

### CODE REVIEW

**Comment by cizezsy** [comment]
Thanks~will look this as soon as possible :P

File: src/main/resources/io/jenkins/plugins/coverage/CoveragePublisher/config.jelly [hunk 1]
```diff
@@ -19,9 +19,11 @@
         <f:entry field="failNoReports" title="Failed if No Reports Found">
             <f:checkbox/>
         </f:entry>
-        <f:entry field="calculateDiffForChangeRequests" title="Calculate diff from change requests">
-            <f:checkbox/>
-        </f:entry>
+        <f:optionalBlock field="calculateDiffForChangeRequests" title="Calculate diff from change requests" inline="true">
+            <f:entry field="failBuildIfCoverageDecreasedInChangeRequest" title="Fail build if code coverage decreased in change request build">
```
Comment (cizezsy) [inline src/main/resources/io/jenkins/plugins/coverage/CoveragePublisher/config.jelly]: Maybe we need to move the description to the help file(like [this](https://github.com/jenkinsci/code-coverage-api-plugin/blob/dev/src/main/resources/io/jenkins/plugins/coverage/CoveragePublisher/help-calculateDiffForChangeRequests.html)) and make the title shorter?

How to add a help file for form field: https://wiki.jenkins.io/display/JENKINS/Basic+guide+to+Jelly+usage+in+Jenkins#BasicguidetoJellyusageinJenkins-Helpfiles

File: src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java [hunk 2]
```diff
@@ -122,6 +124,9 @@ public void performCoverageReport(List<CoverageReportAdapter> reportAdapters, Li
 
         HealthReport healthReport = processThresholds(results, globalThresholds, action);
         action.setHealthReport(healthReport);
+        if (failBuildIfCoverageDecreasedInChangeRequest) {
```
[the remark this answers is not in the document]
Comment (tempora-mutantur) [reply src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java]: Thank you, resolved.

File: src/main/resources/io/jenkins/plugins/coverage/CoveragePublisher/config.jelly (hunk 1 above)
Comment (tempora-mutantur) [reply src/main/resources/io/jenkins/plugins/coverage/CoveragePublisher/config.jelly]: image


Thank you

**Comment by cizezsy** [comment]
merged it, thanks!

**Comment by gary287** [comment]
I would really like to use this new option. When do you plan on releasing an updated version of the plugin?

**Comment by cizezsy** [comment]
@gary287 released this in version 1.1.1 :)

**Comment by gary287** [comment]
Thank you!

**Comment by thatsIch** [comment]
How do you use this option in Jenkins with declarative pipeline syntax?
Is there some kind of documentation I am missing?

**Pull request merged**